src/main/kotlin/org/rust/cargo/runconfig/CargoRunStateBase.kt

```diff
@@ -22,6 +22,7 @@ import org.rust.cargo.toolchain.RustToolchain
 
 abstract class CargoRunStateBase(
     environment: ExecutionEnvironment,
+    val runConfiguration: CargoCommandConfiguration,
```

Comment:
I'm not sure that we need to pass both `runConfiguration` and `config`, because the `config` can be created from `runConfiguration`. I know that's not entirely true for `buildTests` function, but in this case we can create a new run configuration (with `--no-run` flag) to pass it to the constructor.